wordpress-mobile/WordPress-iOS
# Swift App Delegate 2: Debugging / Logging

This PR converts some of the App Delegate's debugging and logging related methods over to Swift.

I think there are definitely some places the code could be improved now it's in Swift, but for this first step I'd like us to just get everything ported over as closely as possible.

I had to add a temporary extra method to the existing app delegate to set the `ddLogLevel` – I wasn't sure of another way to do this from Swift.

I've also removed `removeCredentialsForDebug`, as @koke confirmed that it wasn't necessary.

To test:

* Ensure that all of the debug info is logged out correctly on launch (it'll begin with `======...`). I actually think this has been missing from the app for a while – I don't recall seeing it for some time. Check that none of the values are logged as `Optional(...)`.
* When logged into the app, navigate into Me > Help & Support and enable Extra Debug
   * Put some breakpoints in `toggleExtraDebuggingIfNeeded`, and ensure that Extra Debug remains enabled when you log out and when you log back in.
   * Repeat the process, but have Extra Debug _disabled_ before you log out. Check that it gets enabled when you log out, and disabled again when you log back in.
* Check that DDLog still works as expected!
